### PR TITLE
[DESK-384] windows code signing

### PR DESCRIPTION
**Pull request by JamieRuderman**
### Description

Made buildable on windows machine, along with full app signing using the EV signing cert.

### Checklist

- [x] Pull Request title is in the format `[PROJ-12345] Some useful description here...`
- [x] ~~I have prefixed this PR with `[WIP]` if it is a Work In Progress (not ready to merge)~~
- [x] The git branch is in the format `PROJ-12345-some-useful-description-here`
- [x] I have added one or more reviewers
- [x] ~~I have added/updated tests for any code changes~~
- [x] I have updated documentation (including README, inline comments and docs.remote.it docs)
- [x] This PR only contains one isolated change (bug fix, feature, etc)
- [x] ~~Translation strings added/updated for all user-visible text~~
- [x] I have manually reviewed this PR to make sure only the files I intended to change were changed and nothing else
- [x] I have notified reviewers on Slack by @ mentioning them in the `#desktop`channel
- [x] I have moved the Jira ticket into `Resolved` state and made a note in the ticket with a link to this PR

### Test plan

On Windows.

1. Install another alpha release
2. Wait for update notice
3. Update 

### Ticket

<https://remoteit.atlassian.net/browse/DESK-384>

### CODE REVIEW

File: scripts/sign.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+  if (conf.hash === 'sha1') return
+  if (require('os').platform() !== 'win32') {
+    console.log(`SKIPPING WINDOWS SIGNING ${conf.path}`)
+    return
+  }
+
+  console.log(`Signing Windows Binary ${conf.path}`)
+
+  require('child_process').execSync(
```
Comment (danawoodman) [inline scripts/sign.js]: What are you requiring here rather than the top of the file?

File: scripts/sign.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  // rather create environment variables with sensitive data
+  const tokenPassword = process.env.CSC_KEY_PASSWORD
+
+  if (conf.hash === 'sha1') return
+  if (require('os').platform() !== 'win32') {
+    console.log(`SKIPPING WINDOWS SIGNING ${conf.path}`)
+    return
+  }
+
+  console.log(`Signing Windows Binary ${conf.path}`)
```
Comment (danawoodman) [inline scripts/sign.js]: No important but this is probably cleaner:

```suggestion
  console.log('Signing Windows Binary', conf.path)
```

File: scripts/sign.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+require('dotenv').config()
+
+exports.default = async function(conf) {
+  // do not include passwords or other sensitive data in the file
+  // rather create environment variables with sensitive data
+  const tokenPassword = process.env.CSC_KEY_PASSWORD
```
Comment (danawoodman) [inline scripts/sign.js]: Nit: maybe throw if this key isn't set?

File: scripts/sign.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+  if (conf.hash === 'sha1') return
+  if (require('os').platform() !== 'win32') {
```
Comment (danawoodman) [inline scripts/sign.js]: Why requiring here vs import?

File: scripts/sign.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    console.log(`SKIPPING WINDOWS SIGNING ${conf.path}`)
```
Comment (danawoodman) [inline scripts/sign.js]:
```suggestion
    console.log('SKIPPING WINDOWS SIGNING', conf.path)
```

**Comment by JamieRuderman** [comment]
The sign.js script was an attempt to get the builder to not ask for the password when building. However, I couldn't get it to work, and upon further investigation found that you cannot get an EV Signing Certificate to build without prompting. There was a workaround that the cert software has a checkbox to only ask once per login for the password, so I added that to the docs. I didn't want to throw away the signing script however, so I left it, kinda incomplete. I will add a comment to the top indicating that it isn't used for signing at this point, but could be if we ever move to a standard Signing Cert so that we can build in an automated fashion.

**Pull request merged**